# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/debug/envChain_function-calls.js
# Commit: d81caf9ce78a
# Full Hash: d81caf9ce78a11c3bb82900a4b6b07dddbabd80e
# Author: Sandor Molnar <smolnar@mozilla.com>
# Date: 2025-07-04 09:11:29
# Regressor Bug: 1970388
# File Overlap Count: 1
# Description:
#   Revert "Bug 1970388 - Part 2: Reflect the enclosing environment to MissingEnvironmentMap. r=bthrall" for causing bug 1975376
#   
#   This reverts commit beba2f2d7c6ccb71e99f40346c9a8f55415ac442.
#   
#   Revert "Bug 1970388 - Part 1: Add more info to EnvironmentObject::dump. r=bthrall"
# ==============================================================================

diff -r c83184b73e25 -r d81caf9ce78a js/src/jit-test/tests/debug/envChain_function-calls.js
--- a/js/src/jit-test/tests/debug/envChain_function-calls.js	Fri Jul 04 04:01:01 2025 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,65 +0,0 @@
-var g = newGlobal({ newCompartment: true });
-g.eval(`
-let counter = 10;
-
-function func1() {
-  var local1 = counter;
-
-  var func2 = function f2() {
-    var local2 = counter;
-
-    var func3 = function f3() {
-      var local3 = counter;
-
-      var func4 = function f4() {
-        var local4 = counter;
-
-        // Close these local variables over to make them available in
-        // the environment objects.
-        var x = [local1, local2, local3];
-
-        counter++;
-
-        debugger;
-      };
-
-      func4();
-    };
-
-    func3();
-  };
-
-  func2();
-};
-`);
-
-var dbg = new Debugger(g);
-
-var result = null;
-
-dbg.onDebuggerStatement = function handleDebuggerStatement(f) {
-  result = {
-    local1: f.eval("local1").return,
-    local2: f.eval("local2").return,
-    local3: f.eval("local3").return,
-    local4: f.eval("local4").return,
-  };
-};
-
-g.eval('func1()');
-
-// All references, with/without the function call bounrary, should see
-// the same value.
-assertEq(result.local1, 10);
-assertEq(result.local2, 10);
-assertEq(result.local3, 10);
-assertEq(result.local4, 10);
-
-g.eval('func1()');
-
-// All references, with/without the function call bounrary, should see
-// the same updated value in the 2nd call's environment chain.
-assertEq(result.local1, 11);
-assertEq(result.local2, 11);
-assertEq(result.local3, 11);
-assertEq(result.local4, 11);